### PR TITLE
Fix issue #8

### PR DESCRIPTION
**Pull request by dardo82**
Add manufacturer's name to the "tree" option, optionally using parenthesis.

### CODE REVIEW

File: lsusb [hunk 1]
```diff
@@ -10,78 +10,77 @@ help ()  {
...
-		# The PID depends on the speed of the hub, which we deduce from the controller driver.
+		# The product ID depends on the speed of the hub, which we deduce from the controller's driver.
 		PID=`echo "$device" | egrep "Controller Driver" | awk -F':' '{print $2}' | sed -e 's/0x//; s/^ *//g' -e 's/ *$//g; s/.*\(....\)$/\1/'`
 	  	case "$PID" in
-			OHCI) PID="8005"; name="OHCI Root Hub Simulation"; speed="12M" ;;
-			EHCI) PID="8006"; name="EHCI Root Hub Simulation"; speed="480M" ;;
-			XHCI) PID="8007"; name="XHCI Root Hub USB 2.0 Simulation"; speed="5000M" ;;
+			OHCI) PID="0001"; name="1.1 root hub"; speed="12M" ;;
+			EHCI) PID="0002"; name="2.0 root hub"; speed="480M" ;;
+			XHCI) PID="0003"; name="3.0 root hub"; speed="5000M" ;;
```
Comment (jlhonora) [inline lsusb:83]: I'm not sure about this change. It is more readable, but the spirit of this script is to emulate Linux's functionality as close as possible. I'll check with a real Linux device and compare it.

File: lsusb [hunk 2]
```diff
@@ -147,20 +146,14 @@ parse ()  {
...
-		if [ -n "$arg_dev" ]; then
-			if [ "$arg_dev" -ne "$device_num" ]; then
-				return 1
-			fi
-		fi
+	   else
+		# A missing colon is a syntax error in -s option.
+    		help
+		exit 1
+	   fi
```
Comment (jlhonora) [inline lsusb:153]: @jlhonora reminder: check this

**Comment by jlhonora** [comment]
Looks well. I'll check it with a linux device and let you know.

**Comment by dardo82** [comment]
It's more congruent internally with this change, even if maybe it's slightly different from the Linux version.

**Comment by dardo82** [comment]
How long will it take to let me know? -_-

**Comment by jlhonora** [comment]
Sorry, totally forgot about this.

I think it needs a lot of rework to be merged. Once again, the spirit of this command is to emulate it as close as possible to the Linux one, but for OS X. Why is this? Some programs or scripts that were ported from Linux to OS X need the exact same behavior.

So:
1. The options for `lsusb -h` are now printed in different order. `-m` should be just above `--version`
2. Spelling corrections in `lsusb -h` don't apply, they're actually that way in Linux. As a reference:
   
   ```
   Usage: lsusb [options]...
   List USB devices
     -v, --verbose
         Increase verbosity (show descriptors)
     -s [[bus]:][devnum]
         Show only devices with specified device and/or
         bus numbers (in decimal)
     -d vendor:[product]
         Show only devices with the specified vendor and
         product ID numbers (in hexadecimal)
     -D device
         Selects which device lsusb will examine
     -t
         Dump the physical USB device hierarchy as a tree
     -V, --version
         Show version of program
   ```
3. Missing collon after `-s` is not an error, current behavior works as expected
4. Hub speed rename isn't appropriate either.

If you want to review these and submit a new PR I'd gladly take a look again. This time won't take long, I promise :)

**Comment by dardo82** [comment]
Can you also give me the output of `lsusb` on a Linux system as opposed to `lsusb -h` for reference ?

**Comment by jlhonora** [comment]
On linux:

```
$ lsusb -h
lsusb: invalid option -- 'h'
Usage: lsusb [options]...
List USB devices
  -v, --verbose
      Increase verbosity (show descriptors)
  -s [[bus]:][devnum]
      Show only devices with specified device and/or
      bus numbers (in decimal)
  -d vendor:[product]
      Show only devices with the specified vendor and
      product ID numbers (in hexadecimal)
  -D device
      Selects which device lsusb will examine
  -t
      Dump the physical USB device hierarchy as a tree
  -V, --version
      Show version of program
```

```
$ lsusb
Bus 001 Device 002: ID 0424:9512 Standard Microsystems Corp. 
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 003: ID 0424:ec00 Standard Microsystems Corp. 
Bus 001 Device 004: ID 0bda:8176 Realtek Semiconductor Corp. RTL8188CUS 802.11n WLAN Adapter
```

**Comment by dardo82** [comment]
I don't have any script or program that needs this, i was doing this just for fun, but if i'm so limited in the ways i can improve it, i think that i will just alias system_profiler SPUSBDataType to `lsusb` and that will suit my tastes, at least for now.

**Pull request closed without merging**